integration/benchmark/cmd/ci/main.go

```diff
@@ -271,3 +298,96 @@ func sendDataToBigQuery(
 	}
 	return nil
 }
+
+// adjustTPS tries to find the maximum TPS that the network can handle using a simple AIMD algorithm.
+// The algorithm starts with minTPS as a target.  Each time it is able to reach the target TPS, it
+// increases the target by `additiveIncrease`. Each time it fails to reach the target TPS, it decreases
+// the target by `multiplicativeDecrease` factor.
+//
+// To avoid oscillation and speedup conversion we skip the adjustment stage if TPS grew
+// compared to the last round.
+//
+// Target TPS is always bounded by [minTPS, maxTPS].
+func adjustTPS(
+	lg *benchmark.ContLoadGenerator,
+	log zerolog.Logger,
+	interval time.Duration,
+	minTPS, maxTPS, maxInflight uint,
+) error {
+	targetTPS := minTPS
+	lastTs := time.Now()
+	lastTPS := float64(minTPS)
+	lastTxs := uint(lg.GetTxExecuted())
+	for {
+		select {
+		case <-time.After(interval):
```

Comment:
use a ticker?

Reply:
ticker is tricky since SetTPS is a blocking call which means I would need to reset the ticker on every run, so I went with a manual `time.After`.
(Without it I would sometimes get "doubleticks" if TPS adjustment took too long.)

Reply:
fair enough.  that's useful context.  maybe add a comment